Added docs for Auth/Proxy LB configuration (#23154)
gravitational/teleport: docs/pages/reference/backends.mdx

## Auth Server State

> To run multiple instances of Teleport Auth Server, you must switch to a High
> Availability secrets back-end first.

Reviewer: ```suggestion
To run multiple instances of Teleport Auth Server, you must switch to a high-availability secrets backend first.
```
- high availability is not a proper noun, so it technically shouldn't be capitalized (though I do see it capitalized elsewhere on this page, so it's probably fine to leave it capitalized here for consistency)
- the rest of the page uses "backend"

> To do this, use a load balancer to create a single auth API access point (AP)

Reviewer: Does "auth API access point" mean the same thing as "a single access point for the Auth Service API"?

and specify this AP in the `auth_server` field of the Teleport configuration

> file for all Nodes in a cluster.

Reviewer: Does "Nodes" mean "Teleport services"? The [Core Concepts](https://goteleport.com/docs/core-concepts/) page doesn't use the "Nodes" term, and we're moving away from it in the docs.

Configure your load balancer to use Layer 4 (TCP) load balancing with a 300
second idle timeout.

<Admonition type="tip" title="NOTE">

> With multiple instances of the auth servers running, special attention needs to

Reviewer: ```suggestion
With multiple instances of the Auth Service running, special attention needs to
```

be paid to keeping their configuration identical. Settings like `cluster_name`,

> `tokens`, `storage`, etc must be the same.

Reviewer: ```suggestion
`tokens`, `storage`, etc. must be the same.
```

</Admonition>

## Proxy State

The Teleport Proxy is stateless which makes running multiple instances trivial.

If using the [default configuration](./networking.mdx), configure your load

> balancer to forward ports `3023`, `3024`, and `3080` to the servers that run

Reviewer: Just double-checking: TLS routing isn't enabled by default?

> the Teleport proxy. If you have configured your proxy to use non-default ports,

Reviewer: ```suggestion
the Teleport Proxy Service. If you have configured your Proxy Service to use non-default ports,
```

## Auth Server State

> To run multiple instances of Teleport Auth Server, you must switch to one of

Reviewer: ```suggestion
To run multiple instances of the Teleport Auth Service, you must switch to one of
```

the high-availability secrets backend listed below first.

Once you have a high-availability secrets backend and multiple instances of

> Auth Servers running, you'll need to create a load balancer to evenly

Reviewer: ```suggestion
the Auth Service running, you'll need to create a load balancer to evenly
```

> distribute traffic to all Auth Servers instances and have a single point of

Reviewer: ```suggestion
distribute traffic to all Auth Service instances and have a single point of
```

> entry for all components that need to communicate with Auth Server. Use the

Reviewer: ```suggestion
entry for all components that need to communicate with the Auth Service. Use the
```
